sdks/python/apache_beam/io/iobase.py

```diff
@@ -886,12 +888,18 @@ def get_desired_chunk_size(total_size):
 
   def expand(self, pbegin):
     if isinstance(self.source, BoundedSource):
+      coders.registry.register_coder(BoundedSource, _MemoizingPickleCoder)
       display_data = self.source.display_data() or {}
       display_data['source'] = self.source.__class__
+
+      def output_source(_) -> BoundedSource:
+        assert isinstance(self.source, BoundedSource)
```

Comment:
Isn't this assert redundant with the if statement a couple lines above?

Reply:
removed the function. (fwiw - I had used the assert to work around mypy complaining about types)